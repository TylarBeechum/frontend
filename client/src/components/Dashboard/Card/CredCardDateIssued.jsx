import React from 'react';
import styled from 'styled-components';

const CredCardDateIssued = () => {
  return (
    <DateIssuedContainer>
      <p>2/2/2018</p>
    </DateIssuedContainer>
  );
};

const DateIssuedContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

export default CredCardDateIssued;
