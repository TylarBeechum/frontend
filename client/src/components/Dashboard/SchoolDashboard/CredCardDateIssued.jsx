import React from 'react'

const CredCardDateIssued = () => {
  return (
    <div>
    <p>2/2/2018</p>
  </div>
  )
}

export default CredCardDateIssued
