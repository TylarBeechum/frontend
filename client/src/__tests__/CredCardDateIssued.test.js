/* eslint-disable */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { fireEvent } from '@testing-library/react';
import {
  renderWithRouterAndProviders,
  rendererWithRouterAndProviders
} from '../__testHelpers__';

import CredCardDateIssued from '../components/Dashboard/Card/CredCardDateIssued';

describe('<CredCardDateIssued />', () => {
  describe('should render the CredCard component successfully without crashing', () => {
    it('tests that the component does not crash when it is rendered', () => {
      const helpers = renderWithRouterAndProviders(
        <CredCardDateIssued date={'02/02/2020'} />
      );
    });
  });

  describe('should match the previous snapshot of the component', () => {
    it('tests that the component has not changed and matched the previous snapshot', () => {
      const tree = rendererWithRouterAndProviders(
        <CredCardDateIssued date={'03/03/3030'} />
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe('should successfully render a string to the screen', () => {
    it('tests that the date string is rendered successfully', () => {
      const { getByText } = renderWithRouterAndProviders(
        <CredCardDateIssued date={'testing'} />
      );
      expect(getByText(/testing/i)).toBeDefined();
    });
  });
});
