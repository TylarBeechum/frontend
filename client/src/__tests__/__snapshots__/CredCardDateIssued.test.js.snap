// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CredCardDateIssued /> should match the previous snapshot of the component tests that the component has not changed and matched the previous snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 iDwiuh"
>
  <div
    className="sc-EHOje hvxJwW"
  >
    <p>
      03/03/3030
    </p>
  </div>
</div>
`;
